go/api/go_api/auth.py

```diff
@@ -47,11 +52,76 @@ def requestAvatarId(self, credentials):
         raise error.UnauthorizedLogin()
 
 
+class GoAuthBouncerCredentialFactory(BasicCredentialFactory):
+    implements(ICredentialFactory)
```

Comment:
It's not a very big change. I'd be happy with a single commit that did it across the files this PR touches already, but I'm also okay with not doing it here.